### PR TITLE
Select field and refactor

### PR DESCRIPTION
**Pull request by max-nes**

### CODE REVIEW

File: src/validators.js [hunk 1]
```diff
@@ -41,6 +41,14 @@ export const documentContainer = value => value instanceof DocumentContainer
 export const noMoreThanAvailableOnBalance = balance => value => {
   return +balance > +value
 }
+export const maxDecimalDigitsCount = maxDecimalDigitsCount => value => {
```
Comment (quinn71) [inline src/validators.js]: mb move to amount validator? or at least use it inside it

File: src/vue/fields/SelectField.vue [hunk 2]
```diff
@@ -18,59 +17,72 @@
...
       </button>
     </div>
-    <p v-if="errorMessage" class="select-field__err-mes">
+    <p
+      v-if="errorMessage"
+      class="select-field__err-mes"
+    >
       {{ errorMessage }}
     </p>
+    <!-- Selected: {{ value }} -->
```
Comment (quinn71) [inline src/vue/fields/SelectField.vue]: sho

File: src/vue/fields/SelectField.vue [hunk 3]
```diff
@@ -84,38 +96,34 @@ export default {
...
+      type: String,
+      default: '',
     },
   },
 
   data: () => ({
-    currentValue: null, // selected item in the list
-    selectedValue: null, // active list element (for arrow navigation)
-    isExpanded: false,
+    selected: null, // selected item in the list
```
Comment (quinn71) [inline src/vue/fields/SelectField.vue]: `selectedItem` // and no comment needed then

File: src/vue/fields/SelectField.vue [hunk 4]
```diff
@@ -84,38 +96,34 @@ export default {
...
+    preSelected: null, // active list element (for arrow navigation)
```
Comment (quinn71) [inline src/vue/fields/SelectField.vue]: `highlightenItem` is amazing here I think

File: src/vue/fields/SelectField.vue [hunk 5]
```diff
@@ -84,38 +96,34 @@ export default {
...
+    isListOpened: false,
     KEY_CODES,
   }),
 
   watch: {
-    isExpanded (value) {
+    isListOpened (value) {
```
Comment (quinn71) [inline src/vue/fields/SelectField.vue]: I'd rather place this stuff explicitly in the `openList`

File: src/vue/fields/SelectField.vue [hunk 6]
```diff
@@ -84,38 +96,34 @@ export default {
...
-      type: Boolean,
-      required: false,
-      default: false,
-    },
     errorMessage: {
       type: String,
-      default: undefined,
+      default: '',
+    },
+    keyAsValueText: {
```
Comment (quinn71) [inline src/vue/fields/SelectField.vue]: i can't figure out what does it serves for

File: src/vue/fields/SelectField.vue [hunk 7]
```diff
@@ -18,59 +17,72 @@
...
 
 export default {
   name: 'select-field',
+
+  filters: {
+    getOptionText (item, keyAsValueText) {
+      const result = keyAsValueText
+        ? _get(item, keyAsValueText, item)
+        : item
+      return typeof result === 'function'
```
Comment (quinn71) [inline src/vue/fields/SelectField.vue]: pls gimme a doc on how to use the component

Comment (quinn71) [reply src/vue/fields/SelectField.vue]: too many complex stuff I don't want to understand

File: src/vue/mixins/form.mixin.js [hunk 8]
```diff
@@ -93,9 +94,24 @@ export default {
...
     },
     clearFields () {
-      Object.keys(this.form).forEach(key => {
-        this.form[key] = ''
-      })
+      this.clearFieldsWithOverriding({})
+    },
+    clearFieldsWithOverriding (overriddenFields) {
+      for (const [fieldName, fieldValue] of Object.entries(this.form)) {
+        if (overriddenFields[fieldName]) {
```
Comment (quinn71) [inline src/vue/mixins/form.mixin.js]: you never use the method with `overriddenFields != {}`, so this `if` makes no sense

File: src/vue/fields/SelectField.vue (hunk 2 above)
Comment (max-nes) [reply src/vue/fields/SelectField.vue]: removed

File: src/vue/fields/SelectField.vue (hunk 6 above)
Comment (max-nes) [reply src/vue/fields/SelectField.vue]: described in the doc

File: src/vue/fields/SelectField.vue (hunk 5 above)
Comment (max-nes) [reply src/vue/fields/SelectField.vue]: moved

File: src/vue/fields/SelectField.vue (hunk 7 above)
Comment (max-nes) [reply src/vue/fields/SelectField.vue]: the doc is ready

File: src/vue/mixins/form.mixin.js (hunk 8 above)
Comment (max-nes) [reply src/vue/mixins/form.mixin.js]: yep, currently the only usage of this method is call in clearFields. you can use the method independently when you need to clear all the form fields but some of them to replace with another value, f.e.: you need to clear the form but not the asset selection value

File: src/validators.js (hunk 1 above)
Comment (max-nes) [reply src/validators.js]: maybe in the future

**Pull request merged**